macOS/DuckDuckGo/CrashReports/Model/CrashReportReader.swift

```diff
@@ -17,68 +17,97 @@
+            Bundle.main.vpnProxyExtensionBundleId,
+            Bundle.main.dbpBackgroundAgentBundleId
+        ].compactMap(\.self)
+    }
+
+    private let fileManager: FileManager
+    private let validBundleIdentifierProvider: () -> [String]
+    private let dateProvider: () -> Date
+
+    init(fileManager: FileManager = .default,
+         validBundleIdentifierProvider: @escaping () -> [String] = CrashReportReader.validBundleIdentifiers,
+         dateProvider: @escaping () -> Date = Date.init) {
+        self.fileManager = fileManager
+        self.validBundleIdentifierProvider = validBundleIdentifierProvider
+        self.dateProvider = dateProvider
+    }
 
     func getCrashReports(since lastCheckDate: Date) -> [CrashReport] {
         var allPaths: [URL]
 
         do {
-            allPaths = try FileManager.default.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
+            allPaths = try fileManager.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
         } catch {
             assertionFailure("CrashReportReader: Can't read content of diagnostic reports \(error.localizedDescription)")
             return []
         }
 
         do {
-            let systemPaths = try FileManager.default.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
+            let systemPaths = try fileManager.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
             allPaths.append(contentsOf: systemPaths)
         } catch {
             assertionFailure("Failed to read system crash reports: \(error)")
         }
 
         let filteredPaths = allPaths.filter({
-            isCrashReportPath($0) && belongsToThisApp($0) && isFile(at: $0, newerThan: lastCheckDate)
+            isCrashReportPath($0) && isFile(at: $0, newerThan: lastCheckDate)
         })
 
-        return filteredPaths.compactMap(crashReport(from:))
+        return filteredPaths
+            .compactMap(crashReport(from:))
+            .filter(matchesBundleID)
     }
 
     private func isCrashReportPath(_ path: URL) -> Bool {
         let validExtensions = [LegacyCrashReport.fileExtension, JSONCrashReport.fileExtension]
-        return validExtensions.contains(path.pathExtension)
+        guard validExtensions.contains(path.pathExtension) else {
+            return false
+        }
+
+        let fileName = path.lastPathComponent.lowercased()
+        return fileName.contains("duckduckgo")
     }
 
-    private func belongsToThisApp(_ path: URL) -> Bool {
-        let hasAppPrefix = path.lastPathComponent.hasPrefix(Self.displayName ?? "DuckDuckGo")
-        let hasVPNPrefix = path.lastPathComponent.hasPrefix(Self.vpnExtensionDisplayName)
+    private func matchesBundleID(_ crashReport: CrashReport) -> Bool {
+        guard let bundleID = crashReport.bundleID else {
+            return true
```

Comment:
We only read bundle IDs for JSON files, for legacy crash reports let's just let them through.